staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission/v1/doc.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +k8s:deepcopy-gen=package
+// +k8s:conversion-gen=k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission
+// +k8s:defaulter-gen=TypeMeta
+
+// Package v1 is the v1 version of the API.
+// +groupName=apiserver.config.k8s.io
```

Comment:
why in this package? Why in the admission plugin? Is this for admission or general apiserver?

Reply:
@sttts 

> why in this package?

We had discussion with @liggitt about where to place EncryptionConfiguration and concluded that this is the best place. See: https://github.com/kubernetes/kubernetes/issues/61599#issuecomment-375711161

> Why in the admission plugin?

The only reason is because `apiserver.config.k8s.io` was here from its inception.

> Is this for admission or general apiserver?

It doesn't relate to admission at all. I'm not sure what is the "general apiserver" but it definitely related to api server.

Reply:
> The only reason is because apiserver.config.k8s.io was here from its inception.

We should move it to a more top-level place, e.g. pkg/apis/config.

Reply:
After I moved it to `pkg/apis/componentconfig`, it started failing with:
```
2018/05/10 14:38:48 Inspecting imports under ./vendor/k8s.io/apiserver/...
2018/05/10 14:38:49 - validating imports for 129 packages in the tree
2018/05/10 14:38:49 -- found forbidden imports for k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission:
2018/05/10 14:38:49 --- k8s.io/kubernetes/pkg/apis/componentconfig
2018/05/10 14:38:49 -- found forbidden imports for k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission/install:
2018/05/10 14:38:49 --- k8s.io/kubernetes/pkg/apis/componentconfig/v1
2018/05/10 14:38:49 -- found forbidden imports for k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/options/encryptionconfig:
2018/05/10 14:38:49 --- k8s.io/kubernetes/pkg/apis/componentconfig
2018/05/10 14:38:49 --- k8s.io/kubernetes/pkg/apis/componentconfig/v1
2018/05/10 14:38:49 - FAIL
```

Reply:
apiserver config logically belongs in k8s.io/apiserver. by "top level", I think @sttts meant something like `k8s.io/apiserver/pkg/apis`, etc

Reply:
@liggitt Thanks, Jordan! It looks like that `k8s.io/apiserver/pkg/apis` contains files for publicly available APIs that have REST endpoints and can be consumed by end users. If it's true, then I don't see a reason for putting component' configuration there.

Also I see that components like `KubeSchedulerConfiguration`, `CloudProviderConfiguration`, `VolumeConfiguration`, etc aren't presented there at all.

So, how the things should be, to make this PR mergeable?

Reply:
@sttts What will be my next step then? Is it "move internal and v1 types of apiserver.config.k8s.io to `vendor/k8s.io/apiserver/pkg/apis/componentconfig`"? (BTW what examples are doing there?!)

Reply:
Yes, move them. Then replace the loading code with a proper codec to internal types and then add a test that both old and new api groups can be loaded.